hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1649,4 +1654,15 @@ private InternalSchema getInternalSchema(TableSchemaResolver schemaUtil) {
       }
     });
   }
+
+  /**
+   * No-op WriteStatusHandler callback. We will return true w/o doing any processing from the callback.
+   */
+  public static class NoOpWriteStatusHandlerCallback implements WriteStatusHandlerCallback {
```

Comment:
can we just make it optional and get rid of this dummy class.

Reply:
Addressed